src/components/tl/ucp/allgather/allgather.c

```diff
@@ -35,6 +35,14 @@ ucc_base_coll_alg_info_t
             {.id   = UCC_TL_UCP_ALLGATHER_ALG_SPARBIT,
              .name = "sparbit",
              .desc = "O(log(N)) SPARBIT algorithm"},
+        [UCC_TL_UCP_ALLGATHER_ALG_LINEAR] =
+            {.id   = UCC_TL_UCP_ALLGATHER_ALG_LINEAR,
+             .name = "linear",
+             .desc = "O(N - 1) Linear algorithm, one-shot"},
+        [UCC_TL_UCP_ALLGATHER_ALG_LINEAR_BATCHED] =
```

Comment:
I am not sure to understand why we need two algorithm: `linear_batched` seems to be strictly more general than `linear`, and coincide with the latter when `allgather_linear_num_posts=auto`
Am I missing something?

Reply:
but the one shot version is the same as the batched version when the parameter `allgather_linear_num_posts=auto`, right? They are not truly different algos, both can be described as the same algo but with a different scalar hyperparameter

Reply:
right, linear is sort of batched but with predefined parameter